spec/content_block_tools/components/content_block_tools/contact_component_spec.rb

```diff
@@ -272,4 +312,132 @@
       end
     end
   end
+
+  describe "when multiple types are present" do
+    let(:details) do
+      {
+        email_addresses:,
+        telephones:,
+        addresses:,
+        contact_links:,
+        description: nil,
+        order:,
+      }
+    end
+
+    let(:component_double) { double(:component, render: "") }
+
+    context "when an order is not given" do
+      let(:order) { nil }
+
+      it "should call the components in the default order" do
+        expect(ContentBlockTools::Contacts::AddressComponent)
+          .to receive(:new).ordered.with(item: addresses.values.first)
+                           .and_return(component_double)
+
+        expect(ContentBlockTools::Contacts::EmailAddressComponent)
+          .to receive(:new).ordered.with(item: email_addresses.values.first)
+                           .and_return(component_double)
+
+        expect(ContentBlockTools::Contacts::TelephoneComponent)
+          .to receive(:new).ordered.with(item: telephones.values.first)
+                           .and_return(component_double)
+
+        expect(ContentBlockTools::Contacts::ContactLinkComponent)
+          .to receive(:new).ordered.with(item: contact_links.values.first)
+                           .and_return(component_double)
+
+        described_class.new(content_block:).render
+      end
+    end
+
+    context "when an order is given" do
+      let(:order) {  %w[telephones.foo addresses.some_address contact_links.foo email_addresses.foo] }
+
+      it "calls the components in the specified order" do
+        expect(ContentBlockTools::Contacts::TelephoneComponent)
+          .to receive(:new).ordered.with(item: telephones.values.first)
+                           .and_return(component_double)
+
+        expect(ContentBlockTools::Contacts::AddressComponent)
+          .to receive(:new).ordered.with(item: addresses.values.first)
+                           .and_return(component_double)
+
+        expect(ContentBlockTools::Contacts::ContactLinkComponent)
+          .to receive(:new).ordered.with(item: contact_links.values.first)
+                           .and_return(component_double)
+
+        expect(ContentBlockTools::Contacts::EmailAddressComponent)
+          .to receive(:new).ordered.with(item: email_addresses.values.first)
+                           .and_return(component_double)
```

Comment:
interesting - I don't think I'd have thought to demonstrate the ordering with the sequence of method calls!